pkg/kv/kvserver/rangefeed/scheduled_processor.go

```diff
@@ -633,6 +654,33 @@ func (p *ScheduledProcessor) enqueueRequest(req request) {
 	}
 }
 
+// unregisterClientAsync enqueues an unregister request without
+// blocking. If the request queue is filled, the overflow bool is set
+// which will trigger a scan of all registrations on the next
+// process() call.
+//
+// We are OK with these being processed out of order since these
+// requests originate from a registration cleanup, so the registration
+// in question is no longer processing event.
+func (p *ScheduledProcessor) unregisterClientAsync(r registration) {
+	select {
+	case <-p.stoppedC:
+		return
+	default:
+	}
+	req := func(ctx context.Context) {
+		p.unregisterClientInternal(ctx, r)
+	}
+	select {
+	case p.requestQueue <- req:
+		p.scheduler.Enqueue(RequestQueued)
+	default:
+		r.setShouldUnregister(true)
+		p.overflowedUnregRequests.Store(true)
+		p.scheduler.Enqueue(RequestQueued)
```

Comment:
We can make this the _only_ way unregister requests are handled and simplify this code further. Then, all the tests for unregistration are also hitting this path.

Reply:
Also, are we messing up the metrics update if we take this path? I don't see us call `RangeFeedRegistrations.Dec(.)` in `unregisterMarkedRegistrations`.
<!-- Reviewable comment -OBPbAux06TPs8tNLqmQ:b-bub0q1 -->
<!-- Sent from Reviewable.io -->